### PR TITLE
🪟🎉 Connector builder UI: User inputs, request options, authentication, pagination, slicing

### PR DESCRIPTION
**Pull request by flash1293**
## What

Add user inputs, request options, authentication, pagination, slicing to the connector builder UI.

## How

This is an integration branch.

See details in these PRs:
* https://github.com/airbytehq/airbyte/pull/20385
* https://github.com/airbytehq/airbyte/pull/20431
* https://github.com/airbytehq/airbyte/pull/20577
* https://github.com/airbytehq/airbyte/pull/20645
* https://github.com/airbytehq/airbyte/pull/20664
* https://github.com/airbytehq/airbyte/pull/20712
* https://github.com/airbytehq/airbyte/pull/20497
* https://github.com/airbytehq/airbyte/pull/20698
* https://github.com/airbytehq/airbyte/pull/20748
* https://github.com/airbytehq/airbyte/pull/20715

Changes directly on this PR:
* https://github.com/airbytehq/airbyte/pull/20809/commits/e0e3796afeb4158368023b433797fa42180854a5 fixing a small merge conflict
* https://github.com/airbytehq/airbyte/pull/20809/commits/e68ff49d3cce424d6ccdd536c5a1d13c8acf8db6 and https://github.com/airbytehq/airbyte/pull/20809/commits/e4f7bba80441fb1ee4a392ea96ba1b5d715cb5c9 for performance reasons - otherwise there is one request per keystroke in the form (also validation is expensive for large connectors so it's debounced as well)
* https://github.com/airbytehq/airbyte/pull/20809/commits/2e20fa6c48e0e0f5ee2fcf1dbab41f33aaff0815 to always disable the test button on form errors (do not limit to the current stream) - this is necessary because an invalid stream breaks the whole manifest so everything needs to be valid to test any given stream
* https://github.com/airbytehq/airbyte/pull/20809/commits/65199d02aca0663ae3907df2385a3f0a9dae1dd1 fixes the type of an input

Unmerged PRs pointing here:
* https://github.com/airbytehq/airbyte/pull/20817
* https://github.com/airbytehq/airbyte/pull/20816
* https://github.com/airbytehq/airbyte/pull/20861
* https://github.com/airbytehq/airbyte/pull/20862
* https://github.com/airbytehq/airbyte/pull/20620

In some cases existing errors behave weird (switching between visible and hidden errors in unexpected ways). I'm not sure what this is caused by, but I would like to split it out to investigate.

### CODE REVIEW

**Review by lmossman — APPROVED** [review]
Had one question and pushed a couple of changes. Otherwise the commits you called out look good to me.

Tested locally on this branch and it appears to be working as expected

**Comment by lmossman** [comment]
Just pushed one more change (https://github.com/airbytehq/airbyte/pull/20809/commits/64a5eb9a90ea9a95ce98ca776c697af61d211252) here to prevent this warning from showing up in the console whenever a request parameter/header key-value pair is added:
```
Warning: You submitted a Formik form using a button with an unspecified `type` attribute.  Most browsers default button elements to `type="submit"`. If this is not a submit button, please add `type="button"`.
```

File: airbyte-webapp/src/components/connectorBuilder/Builder/Builder.tsx [hunk 1]
```diff
@@ -1,31 +1,53 @@
...
+
+export const Builder: React.FC<BuilderProps> = ({ values, toggleYamlEditor, validateForm }) => {
   const { setBuilderFormValues, selectedView } = useConnectorBuilderState();
+  const debouncedSetBuilderFormValues = useMemo(
+    () =>
+      debounce((values) => {
+        // kick off formik validation
+        validateForm();
+        // update upstream state
+        setBuilderFormValues(values, builderFormValidationSchema.isValidSync(values));
```
[the remark this answers is not in the document]
Comment (lmossman) [reply airbyte-webapp/src/components/connectorBuilder/Builder/Builder.tsx:39]: Maybe worth capturing in an issue so we don't forget about this?

[the remark this answers is not in the document]
Comment (flash1293) [reply airbyte-webapp/src/components/connectorBuilder/Builder/Builder.tsx:39]: done

**Pull request merged**